component/configsource.go

```diff
@@ -0,0 +1,80 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// SessionParams is passed to ConfigSource at the begin and end of
+// a configuration session.
+type SessionParams struct {
+	// LoadCount tracks the number of the configuration load session.
+	LoadCount int
+}
+
+// ConfigSource is the interface to be implemented by objects used by the collector
+// to retrieve external configuration information.
+type ConfigSource interface {
+	Component
+
+	// BeginSession signals that the object is about to be used to inject data into
```

Comment:
Which "object" is this referring to? The ConfigSource itself? It is a bit confusing, maybe just say ConfigSource instead?